src/main/java/com/github/princesslana/eriscasper/rx/websocket/RxWebSocket.java

```diff
@@ -51,11 +50,13 @@ private Listener(ObservableEmitter<RxWebSocketEvent> em) {
     @Override
     public void onClosed(WebSocket ws, int code, String reason) {
       em.onNext(ClosedTuple.of(ws, code, reason));
+      em.onComplete();
     }
 
     @Override
     public void onClosing(WebSocket ws, int code, String reason) {
       em.onNext(ClosingTuple.of(ws, code, reason));
+      ws.close(0, "Close.");
```

Comment:
We shouldn't use `Close.` as the reason. Try something having to do with `reason` or some other way of finding out what it should be. ws.close() takes websocket closing params with a reason. I don't see `close.` as a good reason for this in any way. https://docs.oracle.com/javaee/7/api/javax/websocket/CloseReason.CloseCodes.html you could use this as a guide for codes. `0` I don't think is a valid one, especially since we don't know if it's graceful.

Reply:
Even if it is graceful we should pass `reason` through for the reason on ws.close(). I think it's useful for transmitting it to the user rather than letting it dissipate. Also useful for telling the other side why we're closing. `Close.` isn't very informative.

Reply:
Also, the closing statement there doesn't mean the close was graceful, it just means it's closing the socket gracefully. We don't know how graceful the endpoint was in talking back to us. So the reason might be good to pass through. Even if we just send a graceful code as per `1001`.

Reply:
At the network level, it is graceful. The other end has told us their closing the connection, we respond by closing it. There's no network failures or cutoffs or other action we can take.

At the semantic/user of the bot level, maybe it's not, but that's already been communicated back up the stack by emitting the ClosingTuple event.

Reply:
Alright, fair enough, maybe use something like this:
`ws.close(1001, "Shutting down.");`
imo `Close.` isn't good enough in the circumstance, it feels insufficient for what it's asking for.
I don't think `Close.` is a `reason`, idk more of a grammatical issue in a sense, nitpicking ofc.

Reply:
Have updated status code to 1000, 'cos for me this is just the underlying transport behaving like it should, nothing abnormal. I've removed the message since it's optional and adds no info.

Reply:
Fair enough. I think this is a good decision.